metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -264,12 +275,33 @@ class Compilers(
   )(fn: (PresentationCompiler, Position) => T): Option[T] = {
     val path = params.getTextDocument.getUri.toAbsolutePath
     loadCompiler(path, interactiveSemanticdbs).map { compiler =>
-      val input = path
-        .toInputFromBuffers(buffers)
-        .copy(path = params.getTextDocument.getUri())
-      val pos = params.getPosition.toMeta(input)
-      val result = fn(compiler, pos)
-      result
+      def defaultInputPos = {
+        val input = path
+          .toInputFromBuffers(buffers)
+          .copy(path = params.getTextDocument.getUri())
+        val pos = params.getPosition
+        (input, pos)
+      }
+
+      val ammoniteInputPosOpt =
```

Comment:
Move to a separate function and make both private?

Reply:
Done.